File: lib/util/core/src/file_size.rs
pub const fn bytes(v: u64) -> u64 {
	v
}

pub const fn kilobytes(v: u64) -> u64 {
	v * 1000
}

pub const fn megabytes(v: u64) -> u64 {
	v * 1000 * 1000
}

pub const fn gigabytes(v: u64) -> u64 {
	v * 1000 * 1000 * 1000
}

pub const fn kibibytes(v: u64) -> u64 {
	v * 1024
}

pub const fn mebibytes(v: u64) -> u64 {
	v * 1024 * 1024
}

pub const fn gibibytes(v: u64) -> u64 {
	v * 1024 * 1024 * 1024
}

File: frontend/packages/components/src/lib/filesize.ts
export const fileSize = {
	bytes(v: number) {
		return Math.floor(v);
	},

	kilobytes(v: number) {
		return this.bytes(v * 1000);
	},
	megabytes(v: number) {
		return this.bytes(v * 1000 * 1000);
	},
	gigabytes(v: number) {
		return this.bytes(v * 1000 * 1000 * 1000);
	},

	kibibytes(v: number) {
		return this.bytes(v * 1024);
	},
	mebibytes(v: number) {
		return this.bytes(v * 1024 * 1024);
	},
	gibibytes(v: number) {
		return this.bytes(v * 1024 * 1024 * 1024);
	},
};
